src/backend/cuda/genpup.py

```diff
@@ -320,7 +321,16 @@ def switcher(typelist, pupstr, nests):
 ##### main function
 ########################################################################################
 if __name__ == '__main__':
-    #### generate the core pack/unpack kernels
+    ##### parse user arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--pup-max-nesting', type=int, help='maximum nesting levels to generate', required=True)
+    args = parser.parse_args()
```

Comment:
How about add some sanity check so the nesting level is not negative or too big? I think it is nice to have at least a warning when the nesting level is above e.g. 4 so make sure the user is aware of what is going to happen.

Reply:
I can check for negative values.  But "too large" is an arbitrary number, which I don't want to add.